### PR TITLE
Add a nightly lockfile regeneration CI job

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Nightly, a CI job will run to regenerate our lockfile.
- A PR will be created to merge this into main
- If tests pass, the PR will be automatically merged.
- If tests fail, the PR will stay open and visible.

## Testing

- Tested as working in a forked repo

## Docs

- N/A

### CODE REVIEW

**Review by natemoo-re — COMMENTED** [review]
Definitely in favor of this!

Is it worth merging @jonathantneal's dependency updates first?
https://github.com/withastro/astro/pull/2085 and https://github.com/withastro/astro/pull/2087

**Comment by FredKSchott** [comment]
Yes, blocked by #2085 but should be merged in immediately after to keep us up-to-date. Working on #2085 with @jonathantneal now

**Review by natemoo-re — APPROVED** [review]
LGTM! Good to go once https://github.com/withastro/astro/pull/2085 is merged in.

**Pull request merged**
